lib/Runtime/Library/RegexHelper.cpp

```diff
@@ -1261,12 +1257,13 @@ namespace Js
         JavascriptString* newString = nullptr;
         const char16* inputStr = input->GetString();
         CharCount inputLength = input->GetLength();
-        const int rawNumGroups = pattern->NumGroups();
+        const uint16 numGroups = pattern->NumGroups();
         Var nonMatchValue = NonMatchValue(scriptContext, false);
         UnifiedRegex::GroupInfo lastMatch; // initially undefined
 
-        AssertOrFailFast(0 < rawNumGroups && rawNumGroups <= INT16_MAX);
-        const uint16 numGroups = uint16(rawNumGroups);
+        // numGroups is always positive because the entire regular expression
+        // itself counts as a capturing group.
+        AssertOrFailFast(0 < numGroups && numGroups <= INT16_MAX);
```

Comment:
> we limit the number of groups to 2^15-1 in the parser

`2^16-1 == 0x7FFF` would be INT16_MAX. Assuming the above comment is correct, shouldn't we want to assert `<= 2^15-1`?

Reply:
(Spoke to Doug in person; we're all good.)

Reply:
Yeah whoops -- bitwise math :P